tools/swift-format/Sources/Rules/NoAccessLevelOnExtensionDeclaration.swift

```diff
@@ -10,7 +10,148 @@ import SwiftSyntax
 ///         declaration in the extension; declarations with redundant access levels (e.g.
 ///         `internal`, as that is the default access level) have the explicit access level removed.
 ///
+/// TODO: Find a better way to access modifiers and keyword tokens besides casting each declaration
+///
 /// - SeeAlso: https://google.github.io/swift#access-levels
 public final class NoAccessLevelOnExtensionDeclaration: SyntaxFormatRule {
+  
+  public override func visit(_ node: ExtensionDeclSyntax) -> DeclSyntax {
+    guard let modifiers = node.modifiers, modifiers.count != 0 else { return node }
+    
+    var modifierIterator = modifiers.makeIterator()
+    if let accessKeyword = modifierIterator.next() {
+
+      let keywordKind = accessKeyword.name.tokenKind
+      switch keywordKind {
+      // Public, private, or fileprivate keywords need to be moved to members
+      case .publicKeyword, .privateKeyword, .fileprivateKeyword:
+        diagnose(.moveAccessKeyword(keyword: accessKeyword.name.text), on: accessKeyword)
+        let newMembers = SyntaxFactory.makeMemberDeclBlock(
+          leftBrace: node.members.leftBrace,
+          members: addMemberAccessKeywords(memDeclBlock: node.members, keyword: accessKeyword),
+          rightBrace: node.members.rightBrace)
+        return node.withMembers(newMembers)
+                .withModifiers(removeModifier(curModifiers: modifiers, removal: accessKeyword))
+      // Internal keyword redundant, delete
+      case .internalKeyword:
+        diagnose(.removeRedundantAccessKeyword(name: node.extendedType.description), on: accessKeyword)
+        return node.withModifiers(removeModifier(curModifiers: modifiers, removal: accessKeyword))
+                .withExtensionKeyword(node.extensionKeyword.withOneLeadingNewline())
+      default:
+        return node
+      }
+    }
+    return node
+  }
+  
+  // Returns modifier list without the access modifier
+  func removeModifier(curModifiers: ModifierListSyntax, removal: DeclModifierSyntax) -> ModifierListSyntax {
+    var newMods: [DeclModifierSyntax] = []
+    for modifier in curModifiers {
+      if modifier.name != removal.name {
+        newMods.append(modifier)
+      }
+    }
+    return SyntaxFactory.makeModifierList(newMods)
+  }
+  
+  // Adds given keyword to all members in declaration block
+  func addMemberAccessKeywords(memDeclBlock: MemberDeclBlockSyntax, keyword: DeclModifierSyntax) -> MemberDeclListSyntax {
+    var newMembers: [MemberDeclListItemSyntax] = []
+    let formattedKeyword = keyword.withName(keyword.name.withLeadingTrivia(.newlines(1) + .spaces(2)))
```

Comment:
I think you'll need to swap the trivia from the first token in the declaration instead of hard-coding 1 newline + 2 spaces.